tests/Providers/Trustpilot/TrustpilotApiClientTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace PodPoint\Reviews\Tests\Providers\Trustpilot;
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Psr7\Request;
+use PodPoint\Reviews\AccessToken;
+use PodPoint\Reviews\ApiClientInterface;
+use PodPoint\Reviews\Providers\Trustpilot\TrustpilotApiClient;
+use PodPoint\Reviews\Tests\TestCase;
+use Mockery;
+use Psr\Http\Message\ResponseInterface;
+
+class TrustpilotApiClientTest extends TestCase
+{
+    protected $apiClient;
+
+    /**
+     * Setting up test instance.
+     */
+    protected function setUp()
+    {
+        $apiKey = 'api-key-123';
+        $secretKey = 'api-secret-key-123';
+        $username = 'api-username';
+        $password = 'api-password';
+
+        $this->apiClient = new TrustpilotApiClient(
+            $apiKey,
+            $secretKey,
+            $username,
+            $password
+        );
+    }
+
+    /**
+     * Making sure the Trustpilot constructor is setting properties and constructs instance of a valid http client.
+     */
+    public function testConstruct()
+    {
+        $this->assertInstanceOf(ApiClientInterface::class, $this->apiClient);
+
+        $this->assertEquals('api-key-123', $this->apiClient->getApiKey());
+        $this->assertEquals('api-secret-key-123', $this->apiClient->getSecretKey());
+        $this->assertEquals('api-username', $this->apiClient->getUsername());
+        $this->assertEquals('api-password', $this->apiClient->getPassword());
+    }
+
+    /**
+     * Making sure access token can be grabbed and returned as AccessToken model.
+     */
```

Comment:
I think this could throw exception also because getAccessToken function

Reply:
Yes, the access token request required field validation. Which is handled in base request.